digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/vvh/VVHClient.scala

```diff
@@ -492,163 +215,205 @@ trait VVHClientOperations {
+    if (!polygonString.contains("{rings:["))
+    {
+      return  Seq.empty[String]
+    }
+    val nvps = new ArrayList[NameValuePair]()
+    nvps.add(new BasicNameValuePair("layerDefs", layerDefinitionWithoutEncoding("", Some("LINKID"))))
+    nvps.add(new BasicNameValuePair("geometry", polygonString))
+    nvps.add(new BasicNameValuePair("geometryType", "esriGeometryPolygon"))
+    nvps.add(new BasicNameValuePair("spatialRel", "esriSpatialRelIntersects"))
+    nvps.add(new BasicNameValuePair("returnGeometry", "false"))
+    nvps.add(new BasicNameValuePair("returnZ", "true"))
+    nvps.add(new BasicNameValuePair("returnM", "true"))
+    nvps.add(new BasicNameValuePair("geometryPrecision", "3"))
+    nvps.add(new BasicNameValuePair("f", "pjson"))
+
+    fetchVVHFeatures(serviceUrl, nvps) match {
+      case Left(features) => features.map(extractLinkIdFromVVHFeature)
+      case Right(error) => throw new ClientException(error.toString)
+    }
   }
 
   /**
-    * Returns VVH road links.
-    */
+   * Returns VVH road links.
+   */
   protected def queryByLinkIds[T](linkIds: Set[Long],
                                   fieldSelection: Option[String],
                                   fetchGeometry: Boolean,
                                   resultTransition: (Map[String, Any], List[List[Double]]) => T,
                                   filter: Set[Long] => String): Seq[T] = {
-    val batchSize = 100
+    val batchSize = 1000
     val idGroups: List[Set[Long]] = linkIds.grouped(batchSize).toList
     idGroups.par.flatMap { ids =>
       val definition = layerDefinition(filter(ids), fieldSelection)
       val url = serviceUrl(definition, queryParameters(fetchGeometry))
 
       fetchVVHFeatures(url) match {
-        case Right(features) => features.map { feature =>
+        case Left(features) => features.map { feature =>
+          val attributes = extractFeatureAttributes(feature)
+          val geometry = if (fetchGeometry) extractFeatureGeometry(feature) else Nil
+          resultTransition(attributes, geometry)
+        }
+        case Right(error) => throw new ClientException(error.toString)
+      }
+    }.toList
+  }
+
+  /**
+   * Returns VVH road links related with finnish names. TODO only one used
+   */
+  protected def queryByNames[T](names: Set[String],
+                                fieldSelection: Option[String],
+                                fetchGeometry: Boolean,
+                                resultTransition: (Map[String, Any], List[List[Double]]) => T,
+                                filter: Set[String] => String): Seq[T] = {
+    val batchSize = 1000
+    val nameGroups: List[Set[String]] = names.grouped(batchSize).toList
+    nameGroups.par.flatMap { names =>
+      val definition = layerDefinition(filter(names), fieldSelection)
+      val url = serviceUrl(definition, queryParameters(fetchGeometry))
+
+      fetchVVHFeatures(url) match {
+        case Left(features) => features.map { feature =>
```

Comment:
Leftright pitää kääntää toisinpäin.
Leftiin virhe, rightiin oikea vastaus, jos sellainen saadaan

esim.:
https://alvinalexander.com/scala/scala-either-left-right-example-option-some-none-null/